server/src/main/java/org/opensearch/action/search/SearchRequestSlowLog.java

```diff
@@ -160,6 +160,22 @@ void onRequestEnd(SearchPhaseContext context, SearchRequestContext searchRequest
         }
     }
 
+    /**
+     * register this listener to TransportSearchAction
+     */
+    @Override
+    protected void register() {
+        TransportSearchAction.addSearchOperationsListener(this);
```

Comment:
Eep... we don't want to make static calls ever.

A better option is to inject a singleton into `TransportSearchAction` (`SearchRequestListenerManager`?) and inject that same thing into each of the listeners. Then we can share some state between them.

The listeners could register themselves with the manager and the action can ask the manager for registered listeners (possibly passing in some state from the request if individual listeners support per-request enabling/disabling). The manager will return the correct set of listeners to be attached to the action for the current request.